### ISSUE TITLE
LADE loss implementation

### ISSUE THREAD

**Issue by aiexperience** [hyperconnect/LADE#13]
Hi,
I got another question from the issues12# but it's a slightly different question, so I opened one more.

My question is the meaning of the 'PerformanceLoss' and 'RouteWeightLoss' in the code.
Is it a sort of classification between training phase loss and inference (evaluate) phase loss?

For example, to apply the LADE loss we need loss/PriorCELoss.py and loss/LADELoss.py. My understanding is at the training phase, the code said PerformanceLoss (Maybe PriorCELoss) will be applied. At the inference phase, RouteWeightLoss (Maybe LADELoss) will be applied. Is it right?

Code location:
https://github.com/hyperconnect/LADE/blob/150181e/config/CIFAR100_LT/lade.yaml#L2-L11

**Comment by juice500ml** [comment]
We were closely following the implementation of [Causal norm](https://github.com/KaihuaTang/Long-Tailed-Recognition.pytorch), where they implement classification loss (PerformanceLoss) and its regularizer (RouteWeightLoss) like that.

Our implementation has a bit of a misnomer. Currently LADELoss.py implements "LADER (eq. 16 in the paper)" and PriorCELoss.py implements "LADE-CE (eq. 18-19 in the paper). Hence, the linear combination of the two ("LADE (eq. 20)") is the final loss, which is implemented [here](https://github.com/hyperconnect/LADE/blob/150181e447e714475e433e532b09990e920e1a1b/run_networks.py#L264-L277).

Hence, both losses are applied in training!
